### DISCUSSION TITLE
rspamd LANGUAGE FILTER

### DISCUSSION THREAD

**Discussion by Ormatic2** [postalserver/postal#1754] [Development discussions]
Hi, I'm new here and I don't know the forum rules. Sorry for it and my bad english.

I want to share a solution to add a new **filter by Language**.
There is little information on the Internet, and it has been difficult for me to understand how to create a new LUA module.
I want to help other people with the same problem.

This is all based on the Rspamd documentation, but with some customizations.
[https://rspamd.com/doc/lua/examples.html](url)

Debian Linux 10
Rspamd 3.1

This LUA module marks as SPAM only the indicated languages. The original Rspamd module allows only the indicated languages.

1.- Create new file ex: ../local.d/lang_filter.lua

     local rspamd_logger = require 'rspamd_logger'
     
     local deny_langs = {
             ['de'] = true,
             ['ja'] = true,
             ['zh'] = true,
             ['ru'] = true,
     }

     rspamd_config:register_symbol{
       type = 'normal',
       name = 'LANG_FILTER',
       score = 6.0,
       group = 'LANG_FILTER',
       description = 'Deny languages',
       flags = 'fine',
       callback = function(task)
                     local any_ok = false
                     local parts = task:get_text_parts() or {}
                     local ln
                     for i,p in ipairs(parts) do
                             ln = p:get_language() or ''
                             local dash = ln:find('-')
                             if dash then
                                     -- from zh-cn to zh
                                     ln = ln:sub(1, dash-1)
                             end

                             if deny_langs[ln] then
                                     rspamd_logger.infox("lang for %1 is %2 -DENY", i, ln)
                             else
                                     any_ok = true
                                     rspamd_logger.infox("lang for %1 is %2 -OK", i, ln)
                                     break
                             end
                     end
                     if any_ok or not ln or #ln == 0 then
                             return false
                     else
                             return true
                     end
       end,
     }

2.- Create new file ex: ../local.d/rspamd.lua

     local local_conf = rspamd_paths['LOCAL_CONFDIR']
     dofile(local_conf .. '/local.d/lang_filter.lua')

The file "lang_filter.lua" for create new Rule and new Symbol. Put the desired score.
You can see the result in the LOG file.

The file "rspamd.lua" for call de new Rule.
You can also use rspamd.local.lua

I hope I have helped and your feel free to correct and improve this example.

**Comment by willpower232** [comment]
Good to know, thanks for sharing. This will no doubt be useful to someone at some point.
